sbncode/BeamSpillInfoRetriever/BNBRetriever/MWRData.cpp

```diff
@@ -0,0 +1,71 @@
+#include <iostream>
+#include <iomanip>
+#include <string>
+#include <sstream>
+#include <vector>
+#include <boost/algorithm/string.hpp>
+#include "MWRData.h"
+
+using namespace std;
+
+namespace sbn{
+
+  std::vector< std::vector < int > > MWRData::unpackMWR(std::string packed_data, std::vector<double> &time_stamp, double timeoffset) const
+{
+
+  std::vector<std::vector<int> > unpacked_data;
+  unpacked_data.resize(4);
+  short data[444];
+
+  std::vector<std::string> row(0);
+  boost::split(row, packed_data, boost::is_any_of(","));
+  if (row.size()==447) {
+    for (int i=3;i<447;i++) {
+      data[i-3]=atoi(row[i].c_str());
+    }
+    string devname=row[1].substr(0,8);
+    for (int idev=0;idev<4;idev++) {
+      mwrpulse_t mwr=getMWRdata(data,idev);
+      time_stamp.push_back(mwr.sheader.timesec+mwr.sheader.timensec/1000000000.+timeoffset);
+      for (int ich=0;ich<48;ich++) {
+	unpacked_data[idev].push_back(mwr.hor[ich]);
+      }
+      for (int ich=0;ich<48;ich++) {
+	unpacked_data[idev].push_back(mwr.ver[ich]);
+      }
+    }
+  } else {
+    cout <<"Bad data!"<<endl;
```

Comment:
No-one is going to know where this message comes from, if it ever occurs. Should write `BeamSpillInfoRetriever: MRWData:` at the start of it.

Also, I think this needs to be a fatal abort. The consequence of getting here is that we don't understand the spill data we got, but we decide to return (4 empty vectors?) anyway. That presumably means the POT accounting will be off, which can impact the correctness of the results, and the only evidence this has happened is a cryptic printout in a grid log that it's very unlikely anyone ever reads.

I would `abort()` the whole job at this point, since we have no idea how to correctly account for the exposure in this file. Hopefully we never hit this and the point is moot. If it does happen a significant fraction of the time, we can go and look at the types of corrupted data that exist, and decide if there are sensible ways to handle any of them. But we won't even know that's something we need to do unless the production jobs are actively failing.

Reply:
This is just the multiwire data, not what we use for POT accounting (which is the toroid), and there are redundant multiwire systems. I think for the time being letting it report empty data frames is OK and if we see that it is impacting our beam quality cuts then we can revisit if we want this to `abort`

Reply:
I added the requested info to the `cout`